onnxruntime/core/graph/graph.cc

```diff
@@ -2260,6 +2261,9 @@ Status Graph::VerifyNodeAndOpMatch(const ResolveOptions& options) {
   ctx.set_ir_version(gsl::narrow_cast<int>(IrVersion()));
   ctx.set_opset_imports(DomainToVersionMap());
   ctx.set_schema_registry(schema_registry_.get());
+  // Set the parent directory of model path to load external tensors if exist
+  ctx.set_model_dir(ToMBString(ModelPath().ParentPath().ToPathString() + k_preferred_path_separator));
```

Comment:
why it must be ended by '/'?

Reply:
https://github.com/onnx/onnx/blob/bd2ff65b775580d21ab38cb655a04854c348ff49/onnx/checker.cc#L133
It is required here because the code in ONNX does not handle path join...

Reply:
Then onnx should fix it! It is always safe to add an additional path separator there when you do path join. But, it seems a little bit wired to do it here without knowing the result will only be used in path join.

Reply:
And instead of calling std::ifstream, it should use fstat to test if a file exists: https://linux.die.net/man/2/fstat

Reply:
Makes sense. I am proposing a PR in ONNX to fix it before checking this PR in. Thank you for bringing this up.

Reply:
Here is the PR to fix that in ONNX: https://github.com/onnx/onnx/pull/3214